### PR TITLE
Feature/PayPal commit property

### PR DESCRIPTION
**Pull request by dkatashev**
## Summary
The main purpose of this Pull Request is to manage commit status of the transaction - https://developer.paypal.com/docs/checkout/reference/customize-sdk/#commit.

This will expand the possibilities and make possible the next feature - https://developer.paypal.com/docs/checkout/integration-features/confirmation-page/.

### CODE REVIEW

File: packages/lib/src/components/PayPal/Paypal.tsx [hunk 1]
```diff
@@ -28,12 +28,13 @@ class PaypalElement extends UIElement<PayPalElementProps> {
...
     // - if merchant has defined properties directly in props, use these instead
     protected formatProps(props) {
-        const { configuration, intent, merchantId } = props;
+        const { configuration, intent, commit, merchantId } = props;
         return {
             ...props,
             configuration: {
                 ...configuration,
                 ...(intent && { intent }),
+                ...{ commit },
```
Comment (pabloai) [inline packages/lib/src/components/PayPal/Paypal.tsx:37]: No need to add this here. Actually there's no need to change this file at all, as the `commit` property is already included in the props.

File: packages/lib/src/components/PayPal/utils.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ const getPaypalSettings = ({ amount, countryCode, debug, environment = '', local
     const isTestEnvironment: boolean = environment.toLowerCase() === 'test';
     const clientId: string = isTestEnvironment ? ADYEN_CLIENTID_TEST : ADYEN_CLIENTID_LIVE;
 
-    const { merchantId, intent } = configuration;
+    const { merchantId, intent, commit } = configuration;
```
Comment (pabloai) [inline packages/lib/src/components/PayPal/utils.ts:22]: The `commit` property will arribe inside the `props`, not this `configuration` object.

**Review by pabloai — CHANGES_REQUESTED** [review]
Thanks for the contribution!
I added some comments to make it ready to be merged. Cheers.

**Comment by dkatashev** [comment]
Hi @pabloai,
I did the same as the **intent** attribute.
Maybe I didn't fully understand the idea. According to the code - PayPal **configuration** this is the place where such information should be stored, it is also present in unit tests.

Tell me in this case, where is it better to store this attribute?
Thanks in advance

**Comment by pabloai** [comment]
Hi again @dkatashev,

The `intent` property is a special case because we receive it from the backend in the response from the `paymentMethods` endpoint. On this case it should get to the component in the `props` so there's no need of any special handling.

That being said, if I'm not mistaken this `commit` property is used in the Express Checkout flow. Is that what you're aiming to achieve? Would you do the call to the `/payments/details` endpoint later in the flow?

Just trying to understand the use case. Thanks!

**Comment by dkatashev** [comment]
Hi @pabloai,
Thank you for your detailed explanation.
Yes, you're right. I want to make possible PayPal Express Checkout Flow.
I guess I'll be able to initiate a component with that option first. Then on the server get information from PayPal about the customer (payer) based on PayPal Order ID. Then, do the call to the `/payments/details` to confirm the order.
Do you see any problems to make it possible?
I'll be very grateful for your feedback.

File: packages/lib/src/components/PayPal/utils.ts [hunk 3]
```diff
@@ -28,6 +28,7 @@ const getPaypalSettings = ({ amount, countryCode, debug, environment = '', local
         ...(debug && isTestEnvironment && { debug }),
         ...(currency && { currency }),
         ...(intent && { intent }),
+        ...{ commit },
```
Comment (pabloai) [inline packages/lib/src/components/PayPal/utils.ts:31]: There's no need to use the spread operator here. Just using the shorthand is enough.
```suggestion
        commit,
```

**Review by pabloai — COMMENTED** [review]
The PayPal Express Checkout is a flow we are not focusing on (hence the lack of this `commit` prop). I guess it could be done if you store the data instead of doing the `payments/details` call when the `onAdditionalDetails` is called and then do the call once the shopper confirms the order. We can merge this PR so you can test this flow and see if it works for you. I just added a small comment, but it's almost ready to be merged.

File: packages/lib/src/components/PayPal/utils.ts (hunk 3 above)
Comment (dkatashev) [reply packages/lib/src/components/PayPal/utils.ts]: Updated

**Comment by pabloai** [comment]
Merging this PR.
Thanks for your collaboration @dkatashev !

**Review by pabloai — APPROVED** [review]

**Pull request merged**